### PR TITLE
[SPARK-49843][SQL] Fix change comment on char/varchar columns

### PR DESCRIPTION
**Pull request by stefankandic**
### What changes were proposed in this pull request?

Fix the issue in `AlterTableChangeColumnCommand` where changing the comment of a char/varchar column also tries to change the column type to string. 

### Why are the changes needed?

Because the newColumn will always be a `StringType` even when the metadata says that it was originally char/varchar.

### Does this PR introduce _any_ user-facing change?

Yes, the query will no longer fail when using this code path.

### How was this patch tested?

New query in golden files.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by stefankandic** [comment]
@cloud-fan please take a look at this when you get the chance

**Review by MaxGekk — APPROVED** [review]

File: sql/core/src/test/resources/sql-tests/inputs/charvarchar.sql [hunk 1]
```diff
@@ -23,6 +23,7 @@ desc formatted char_tbl1;
 
 alter table char_tbl1 change column c type char(6);
 alter table char_tbl1 change column c type char(5);
+alter table char_tbl1 change column c comment 'text';
```
Comment (MaxGekk) [inline sql/core/src/test/resources/sql-tests/inputs/charvarchar.sql:26]: BTW, as SPARK-49843 says `Fix ... char/varchar columns`, could you check the same for the varchar column `v`

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: sql/core/src/test/resources/sql-tests/inputs/charvarchar.sql (hunk 1 above)
Comment (stefankandic) [reply sql/core/src/test/resources/sql-tests/inputs/charvarchar.sql]: Sure, fixed

**Review by MaxGekk — APPROVED** [review]
Waiting for CI

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @stefankandic.

**Comment by cloud-fan** [comment]
shall we backport the fix to 3.5 as well?

**Review by dongjoon-hyun — COMMENTED** [review]
+1, LGTM. Thank you, @stefankandic and Max.

**Pull request closed without merging**